src/components/HeadCellMenu/MenuList/MenuGroup.tsx

```diff
@@ -0,0 +1,126 @@
+/* eslint-disable import/no-cycle */
+import React, { useState, useRef } from 'react';
+import Typography from '@mui/material/Typography';
+import ArrowRight from '@qlik-trial/sprout/icons/react/ArrowRight';
+import { ExtendedHeadCellMenuItem, MenuItemGroup } from '../types';
+import {
+  StyledMenuItem,
+  StyledListItemIcon,
+  StyledMenuItemLabel,
+  StyledGroupLabel,
+  StyledGreenBorder,
+} from '../styles';
+import RecursiveMenuList from './RecursiveMenuList';
+
+export const interceptClickOnMenuItems = (menuGroups: MenuItemGroup[], cache: SubMenusOpenStatusCache) => {
+  const result = menuGroups.map((group) =>
```

Comment:
Trying to understand what this actually is. It's gets somewhat confusing when you go from `menuGroups` into a `group` that is also a collection of groups?

Reply:
will try to rename properly as i traverse the array